operator/src/main/java/org/bf2/operator/operands/KafkaCluster.java

```diff
@@ -815,19 +826,26 @@ private Storage buildKafkaStorage(ManagedKafka managedKafka, Kafka current, long
         long storageBytes = storagePerBroker;
         storageBytes += calculateSafetyMargin(managedKafka, current);
         storageBytes += calculateFormatOverheadFromFormattedSize(managedKafka, storageBytes);
+        int volumeCount = this.configs.getConfig(managedKafka).getKafka().getVolumeCount();
 
-        PersistentClaimStorageBuilder builder = new PersistentClaimStorageBuilder()
-                .withId(JBOD_VOLUME_ID)
-                .withSize(getAdjustedMaxDataRetentionSize(current, storageBytes).getAmount())
-                .withDeleteClaim(DELETE_CLAIM);
+        List<SingleVolumeStorage> volumes = new ArrayList<>(volumeCount);
+
+        for (int jbodVolumeId = 0; jbodVolumeId < volumeCount; jbodVolumeId++) {
+            PersistentClaimStorageBuilder builder = new PersistentClaimStorageBuilder();
+            builder.withId(jbodVolumeId)
```

Comment:
use the index as volume ID, i.e. 0, 1, 2, ...

Reply:
I don't think we want to expand the scope of this PR to allow the number of volumes to be changed.  That's would be a significant change which would need to be prioritised separately.

With my earlier comment re. factoring in number of storage volumes into the calculation, I just meant just use `JbodStorage#getVolumes().size()` as a factor in the calculation, or perhaps just make it so the function that calculates the number of threads fail if ever that invariant changes:

```
if (JbodStorage#getVolumes().size() > 1) {
  throw new IllegalStateException ("This algorithm doesn't support support > 1 volumes");
} 
```

Reply:
Ah, I see, sorry for misunderstanding!